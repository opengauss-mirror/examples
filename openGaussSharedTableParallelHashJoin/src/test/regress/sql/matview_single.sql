-- prepare
create table test(c1 int);
insert into test values(1);
-- base op
drop Materialized view mvtest_tm;

CREATE MATERIALIZED VIEW mvtest_tm AS select *from test;
select *From mvtest_tm;

insert into test values(1);
REFRESH MATERIALIZED VIEW mvtest_tm;
select *from mvtest_tm;

drop Materialized view mvtest_tm;

-- error
create incremental MATERIALIZED VIEW mvtest_inc AS select *from test;
REFRESH incremental MATERIALIZED VIEW mvtest_tm;

-- test matview with anounymous types
create database test_imv_db with dbcompatibility 'b';
\c test_imv_db

-- case 1: drop type first
create table imv1_t(a set('ad','asf'), c int);
create incremental materialized view imv1_v as select * from imv1_t;
create incremental materialized view imv12_v as select * from imv1_t;
drop type imv1_t_a_set cascade;
select oid, relname from pg_class where relname like 'mlog%';
drop table imv1_t cascade;
select oid, relname from pg_class where relname like 'mlog%';

-- case 2: drop view first
create table imv1_t(a set('ad','asf'), c int);
create incremental materialized view imv1_v as select * from imv1_t;
create incremental materialized view imv12_v as select * from imv1_t;
drop materialized view imv1_v;
select oid, relname from pg_class where relname like 'mlog%';
drop materialized view imv12_v;
select oid, relname from pg_class where relname like 'mlog%';
drop table imv1_t cascade;

-- case 3: drop table directly
create table imv1_t(a set('ad','asf'), c int);
create incremental materialized view imv1_v as select * from imv1_t;
create incremental materialized view imv12_v as select * from imv1_t;
drop table imv1_t cascade;
select oid, relname from pg_class where relname like 'mlog%';

-- case 4: multi columns
create table imv1_t(a set('ad','asf'), b set('b', 'bb'), c int, d set('b', 'bb'));
create incremental materialized view imv1_v as select * from imv1_t;
create incremental materialized view imv12_v as select * from imv1_t;
create incremental materialized view imv13_v as select * from imv1_t;
\d
drop table imv1_t cascade;
select oid, relname from pg_class where relname like 'mlog%';
\d

create table test_syn(id int unique,a1 varchar(20));
create materialized view mv_test_syn as select * from test_syn;
create incremental materialized view imv_test_syn as select * from test_syn;
create synonym s_mv_test_syn for mv_test_syn;
create synonym s_imv_test_syn for imv_test_syn;

REFRESH MATERIALIZED VIEW s_mv_test_syn;
REFRESH MATERIALIZED VIEW s_imv_test_syn;
REFRESH INCREMENTAL MATERIALIZED VIEW s_mv_test_syn;
REFRESH INCREMENTAL MATERIALIZED VIEW s_imv_test_syn;

drop synonym s_mv_test_syn;
drop synonym s_imv_test_syn;
drop table test_syn cascade;

\c regression
drop database test_imv_db;
