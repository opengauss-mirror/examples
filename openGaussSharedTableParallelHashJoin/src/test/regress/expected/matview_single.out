-- prepare
create table test(c1 int);
insert into test values(1);
-- base op
drop Materialized view mvtest_tm;
ERROR:  materialized view "mvtest_tm" does not exist
CREATE MATERIALIZED VIEW mvtest_tm AS select *from test;
select *From mvtest_tm;
 c1 
----
  1
(1 row)

insert into test values(1);
REFRESH MATERIALIZED VIEW mvtest_tm;
select *from mvtest_tm;
 c1 
----
  1
  1
(2 rows)

drop Materialized view mvtest_tm;
-- error
create incremental MATERIALIZED VIEW mvtest_inc AS select *from test;
REFRESH incremental MATERIALIZED VIEW mvtest_tm;
ERROR:  relation "mvtest_tm" does not exist
-- test matview with anounymous types
create database test_imv_db with dbcompatibility 'b';
\c test_imv_db
-- case 1: drop type first
create table imv1_t(a set('ad','asf'), c int);
NOTICE:  CREATE TABLE will create implicit set "imv1_t_a_set" for column "imv1_t.a"
create incremental materialized view imv1_v as select * from imv1_t;
create incremental materialized view imv12_v as select * from imv1_t;
drop type imv1_t_a_set cascade;
NOTICE:  drop cascades to 6 other objects
DETAIL:  drop cascades to table imv1_t column a
drop cascades to materialized view imv1_v
drop cascades to materialized view imv12_v
drop cascades to materialized view imv1_v column a
--?.*
drop cascades to materialized view imv12_v column a
select oid, relname from pg_class where relname like 'mlog%';
 oid | relname 
-----+---------
(0 rows)

drop table imv1_t cascade;
select oid, relname from pg_class where relname like 'mlog%';
 oid | relname 
-----+---------
(0 rows)

-- case 2: drop view first
create table imv1_t(a set('ad','asf'), c int);
NOTICE:  CREATE TABLE will create implicit set "imv1_t_a_set" for column "imv1_t.a"
create incremental materialized view imv1_v as select * from imv1_t;
create incremental materialized view imv12_v as select * from imv1_t;
drop materialized view imv1_v;
select oid, relname from pg_class where relname like 'mlog%';
--?.*
--?.*
--?.*
--?.*
(2 rows)

drop materialized view imv12_v;
select oid, relname from pg_class where relname like 'mlog%';
 oid | relname 
-----+---------
(0 rows)

drop table imv1_t cascade;
-- case 3: drop table directly
create table imv1_t(a set('ad','asf'), c int);
NOTICE:  CREATE TABLE will create implicit set "imv1_t_a_set" for column "imv1_t.a"
create incremental materialized view imv1_v as select * from imv1_t;
create incremental materialized view imv12_v as select * from imv1_t;
drop table imv1_t cascade;
NOTICE:  drop cascades to 3 other objects
--?.*
drop cascades to materialized view imv1_v
drop cascades to materialized view imv12_v
select oid, relname from pg_class where relname like 'mlog%';
 oid | relname 
-----+---------
(0 rows)

-- case 4: multi columns
create table imv1_t(a set('ad','asf'), b set('b', 'bb'), c int, d set('b', 'bb'));
NOTICE:  CREATE TABLE will create implicit set "imv1_t_a_set" for column "imv1_t.a"
NOTICE:  CREATE TABLE will create implicit set "imv1_t_b_set" for column "imv1_t.b"
NOTICE:  CREATE TABLE will create implicit set "imv1_t_d_set" for column "imv1_t.d"
create incremental materialized view imv1_v as select * from imv1_t;
create incremental materialized view imv12_v as select * from imv1_t;
create incremental materialized view imv13_v as select * from imv1_t;
\d
--?.*
--?.*
--?.*
--?.*
--?.*
--?.*
--?.*
(4 rows)

drop table imv1_t cascade;
NOTICE:  drop cascades to 6 other objects
--?.*
--?.*
--?.*
drop cascades to materialized view imv1_v
drop cascades to materialized view imv12_v
drop cascades to materialized view imv13_v
select oid, relname from pg_class where relname like 'mlog%';
 oid | relname 
-----+---------
(0 rows)

\d
           List of relations
 Schema | Name | Type | Owner | Storage 
--------+------+------+-------+---------
(0 rows)

create table test_syn(id int unique,a1 varchar(20));
NOTICE:  CREATE TABLE / UNIQUE will create implicit index "test_syn_id_key" for table "test_syn"
create materialized view mv_test_syn as select * from test_syn;
create incremental materialized view imv_test_syn as select * from test_syn;
create synonym s_mv_test_syn for mv_test_syn;
create synonym s_imv_test_syn for imv_test_syn;
REFRESH MATERIALIZED VIEW s_mv_test_syn;
REFRESH MATERIALIZED VIEW s_imv_test_syn;
REFRESH INCREMENTAL MATERIALIZED VIEW s_mv_test_syn;
ERROR:  "mv_test_syn" is not an incremental materialized view
REFRESH INCREMENTAL MATERIALIZED VIEW s_imv_test_syn;
drop synonym s_mv_test_syn;
drop synonym s_imv_test_syn;
drop table test_syn cascade;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to materialized view mv_test_syn
drop cascades to materialized view imv_test_syn
\c regression
drop database test_imv_db;
